Khan/render-gateway
# Distribute flow types

# Summary

This adds a build step to output `.flow` files alongside our exported JS.

Since we export both `index.js` files and a couple of others from some packages, I have introduced a way to specify these additional export files from within the `index.js` so that we can document exports in a single place that is front and center to the maintained codebase, rather than buried in a hard-coded list in a build tool script. This also prevents us from just exporting a `.flow` file for every single source file.

The build tool reads this info from the `index.js` of each "package" and writes out the files accordingly.

## Issues

This is so that we can rely on types for WEB-1856 work inside the webapp repo.

## Details

To make this easier and also just make code easier to read/maintain, the gateway code is now in its own folder rather than at the root of `src`.

### Test Plan

To test this, I took the following steps.

1. Changed the import in the `run_local.js` file to use `dist` instead of `src`. Flow immediately complains about this because it cannot get the types.
2. Run the package.json command to build the code and types `yarn build`
3. Modify the `.flowconfig` to remove the `dist` folder from the `[Ignore]` section
4. Edit the `run_local.js` to say, pass the `port` as a string and see flow errors now occur
